### PR TITLE
Changed validator for CreateAPluginDialog

### PR DESCRIPTION
**Pull request by drpayyne**
**Description**

Changes the validator for CreateAPluginDialog from a dedicated class to `@FieldValidation` annotation

**Fixed Issues**

1. magento/magento2-phpstorm-plugin#293: Change validation for the new plugin dialog window

**Additional changes**

- Fixed grammar of error messages
- Changed label of field for plugin name: `Name` -> `Plugin Name`

**Contribution checklist**
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with integration/functional tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/CreateAPluginDialogValidator.java [hunk 1]
```diff
@@ -1,259 +0,0 @@
...
-            );
-            JOptionPane.showMessageDialog(
-                    null,
-                    errorMessage,
-                    errorTitle,
-                    JOptionPane.ERROR_MESSAGE
-            );
-
-            return false;
-        }
```
Comment (drpayyne) [inline src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/CreateAPluginDialogValidator.java:255]: This isn't implemented and will be done in a separate PR as per @VitaliyBoyko.

Comment (VitaliyBoyko) [reply src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/CreateAPluginDialogValidator.java:255]: Could you please create an Issue and reference it to this PR. TY

Comment (drpayyne) [reply src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/CreateAPluginDialogValidator.java:255]: Hi @VitaliyBoyko, I've created an issue for the above: https://github.com/magento/magento2-phpstorm-plugin/issues/332. Thanks.

**Comment by VitaliyBoyko** [comment]
Please update the PR. Thanks!

File: src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/CreateAPluginDialogValidator.java (hunk 1 above)
Comment (drpayyne) [reply src/com/magento/idea/magento2plugin/actions/generation/dialog/validator/CreateAPluginDialogValidator.java:255]: Hi @VitaliyBoyko, I've closed the above issue and suggested a better aproach in this comment: https://github.com/magento/magento2-phpstorm-plugin/pull/326#issuecomment-714742389.

**Comment by drpayyne** [comment]
Hi @VitaliyBoyko, I've fixed the merge conflicts. Also, please check https://github.com/magento/magento2-phpstorm-plugin/pull/324#discussion_r510438207.

**Review by VitaliyBoyko — APPROVED** [review]
QA passed ✅
